src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -39,16 +44,38 @@ public SubstringConstraint(string expected) : base(expected)
         }
 
         /// <summary>
+        /// Modify the constraint to ignore case in matching.
+        /// This will call Using(StringComparison.CurrentCultureIgnoreCase) 
+        /// </summary>
+        public override StringConstraint IgnoreCase
+        {
+            get { Using(StringComparison.CurrentCultureIgnoreCase); return base.IgnoreCase; }
+        }
+
+        /// <summary>
         /// Test whether the constraint is satisfied by a given value
         /// </summary>
         /// <param name="actual">The value to be tested</param>
         /// <returns>True for success, false for failure</returns>
         protected override bool Matches(string actual)
         {
-            if (this.caseInsensitive)
-                return actual != null && actual.ToLower().IndexOf(expected.ToLower()) >= 0;
-            else
-                return actual != null && actual.IndexOf(expected) >= 0;
+            if (actual == null) return false;
+
+            var actualComparison = comparisonType ?? StringComparison.CurrentCulture;
+            return actual.IndexOf(expected, actualComparison) >= 0;
+        }
+
+        /// <summary>
+        /// Modify the constraint to the specified comparison.
+        /// </summary>
```

Comment:
My comment must have gotten lost on an earlier commit, but any time we purposely throw an exception throw an exception we need to put a message in the summary saying what the exception is an under what circumstances it will happen. Here's an example: https://github.com/nunit/docs/wiki/Best-practices-for-XML-documentation#exception